### DISCUSSION TITLE
MPV to stay open while changing streams

### DISCUSSION THREAD

**Discussion by 99bobster99** [streamlink/streamlink#3665] [Q&A]
Is there a way to keep the MPV output window open, while switching streams?

The `--idle=yes` command on the command line did not work, nor did the `idle=yes` in the mpv.conf file.

I added this as an issue over on MPV, they mentioned to use mpv.com, which didn't work either. In contrast, VLC stays open.

I'm hoping that there might be some tricks in Streamlink that might make this doable.

**Comment by bastimeyer** [comment]
> keep the MPV output window open, while switching streams

This is not how Streamlink is designed and you can't just "switch streams". It'll always launch a new player process while using the default stream transport options. You'll have to launch mpv separately, create a custom named pipe, use it as player input and let the different Streamlink processes write to it one after another. Or create multiple named pipes and create a playlist with each of them as a new entry.

**Comment by 99bobster99** [comment]
I created an application where I had FFMPEG piped directly to mpv.exe. the problem was that the  mpv output window did not showup (the process was running and was accepting the pipe stream, since the mpv process memory size was changing as I piped buffer contents to it). When I tried the same thing with VLC, the output window showed properly and the video buffers streamed properly. I really like mpv and would rather use it over VLC (due to intermittent horizontal pixel jitter in VLC). Do you have any ideas? I'm currently trying to create an application using mpv.net to see if it works better, since it has a built in streamer.

**Reply by bastimeyer** [reply]
https://mpv.io/manual/stable/#options-force-window

**Reply by 99bobster99** [reply]
Interesting! Using `--force-window=yes` keeps the mpv output window open, but it doesn't show the video stream being piped to it's stdIn from ffmpeg (substituting vlc for mpv, the video stream plays fine). If I don't redirect mpv's stdin to the pipe, streaming works fine between ffmpeg and mpv using `ffmpeg ... - | mpv -`.

It seems that mpv.exe doesn't like to have it's stdin redirected directly, while vlc is okay with it (my application locks up since mpv is not accepting the input onto it's stdIn).

It looks like others are seeing a similar issue with mpv, it may having something to do with the way mpv accepts terminal inputs as well as video on stdIn  ->  https://github.com/mpv-player/mpv/issues/5005

The hunt for a solution continues ... do you have any other suggestions?
